geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java

```diff
@@ -89,7 +89,7 @@
    * 
    * @since GemFire 6.0
    */
-  public ResultCollector<?, ?> execute(String functionId) throws FunctionException;
+  public <T, S> ResultCollector<T, S> execute(String functionId) throws FunctionException;
```

Comment:
Since this is a public API, you should describe the params in javadoc when it's not obvious what T and S refer to.  You can use the `@param` tag as seen in [1].

[1] https://github.com/apache/geode/blob/8bf39571471642beaaa36c9626a61a90bd3803c2/geode-core/src/main/java/org/apache/geode/cache/snapshot/RegionSnapshotService.java